generator/internal/genclient/translator/openapi/openapi_test.go

```diff
@@ -0,0 +1,341 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/googleapis/google-cloud-rust/generator/internal/genclient"
+)
+
+func TestMakeAPI(t *testing.T) {
+	tDir := t.TempDir()
+
+	contents := []byte(testDocument)
+	translator, err := NewTranslator(contents, &Options{
+		Language:    "rust",
+		OutDir:      tDir,
+		TemplateDir: "not used",
+	})
+	if err != nil {
+		t.Errorf("Error in NewTranslator() %q", err)
+	}
+
+	api, err := translator.makeAPI()
+	if err != nil {
+		t.Errorf("Error in makeAPI() %q", err)
+	}
+
+	checkMessage(t, *api.Messages[0], genclient.Message{
+		Documentation: "A resource that represents a Google Cloud location.",
+		Name:          "Location",
+		Fields: []*genclient.Field{
+			{Name: "name",
+				Documentation: "Resource name for the location, which may vary between implementations.",
+				Typez:         genclient.STRING_TYPE,
+				Optional:      true},
+			{Name: "locationId",
+				Documentation: `The canonical id for this location.`,
+				Typez:         genclient.STRING_TYPE,
+				Optional:      true},
+			{Name: "displayName",
+				Documentation: `The friendly name for this location, typically a nearby city name.`,
+				Typez:         genclient.STRING_TYPE,
+				Optional:      true},
+			{Name: "labels",
+				Documentation: "Cross-service attributes for the location.",
+				Typez:         genclient.MESSAGE_TYPE,
+				Optional:      true},
+			{Name: "metadata",
+				Documentation: `Service-specific metadata. For example the available capacity at the given location.`,
+				Typez:         genclient.MESSAGE_TYPE,
+				Optional:      true},
+		},
+	})
+
+	checkMessage(t, *api.Messages[1], genclient.Message{
+		Documentation: "The response message for Locations.ListLocations.",
+		Name:          "ListLocationsResponse",
+		Fields: []*genclient.Field{
+			{Documentation: "A list of locations that matches the specified filter in the request.",
+				Name:     "locations",
+				Typez:    genclient.MESSAGE_TYPE,
+				Repeated: true},
+			{Documentation: "The standard List next-page token.",
+				Name:     "nextPageToken",
+				Typez:    genclient.STRING_TYPE,
+				Optional: true},
+		},
+	})
+}
+
+func checkMessage(t *testing.T, got genclient.Message, want genclient.Message) {
+	if want.Name != got.Name {
+		t.Errorf("Mismatched message name, got=%q, want=%q", got.Name, want.Name)
+	}
+	if want.Documentation != got.Documentation {
```

Comment:
I would use `cmp.Diff` here so that you can see the diff

```
if diff := cmp.Diff(want.Documentation, got.Documentation) {
     t.Errorf("mismatch (-want +got):\n%s", diff)
}
```